x/concentrated-liquidity/types/pool_hooks.go

```diff
@@ -10,13 +12,37 @@ import (
 
 // Action prefixes for pool actions
 const (
-	CreatePositionPrefix     = "createPosition"
-	AddToPositionPrefix      = "addToPosition"
-	WithdrawPositionPrefix   = "withdrawPosition"
-	SwapExactAmountInPrefix  = "swapExactAmountIn"
-	SwapExactAmountOutPrefix = "swapExactAmountOut"
+	CreatePositionPrefix     = "CreatePosition"
+	WithdrawPositionPrefix   = "WithdrawPosition"
+	SwapExactAmountInPrefix  = "SwapExactAmountIn"
+	SwapExactAmountOutPrefix = "SwapExactAmountOut"
 )
 
+// Helper function to generate before action prefix
```

Comment:
Just curious, what are the contracts we are expecting to register for these hooks?

Reply:
Well, the design space is incredibly broad as this is essentially an expanded version of uni v4 hooks. The immediate trigger was KYC pools by provenance but there are many examples listed [here](https://blog.uniswap.org/uniswap-v4)